scene/gui/tab_bar.cpp

```diff
@@ -640,7 +644,19 @@ int TabBar::get_tab_count() const {
 }
 
 void TabBar::set_current_tab(int p_current) {
-	ERR_FAIL_INDEX(p_current, get_tab_count());
+	if (p_current == -1) {
+		// An index of -1 is only valid if deselecting is enabled or there are no valid tabs.
+		bool no_valid_tabs = true;
+		for (int i = 0; i < get_tab_count(); i++) {
+			if (!is_tab_disabled(i) && !is_tab_hidden(i)) {
+				no_valid_tabs = false;
+				break;
+			}
+		}
+		ERR_FAIL_COND_MSG(!deselect_enabled && !tabs.is_empty() && !no_valid_tabs, "Cannot deselect tabs, deselection is not enabled.");
```

Comment:
```suggestion
		ERR_FAIL_COND_MSG(!deselect_enabled && !no_valid_tabs, "Cannot deselect tabs, deselection is not enabled.");
```
`no_valid_tabs` already handles it.